### ISSUE TITLE
add support for `weighted_quantile_sketch` on given batch

### ISSUE THREAD

**Issue by RyanK0902** [ThrunGroup/improving-xgboost#1]
After bin values are found, they are fixed until termination of the node-splitting. 
Identify where these bin values are stored and how to access the bin values. And then need to identify the start/end points of this logic so we can modularize by iteration.

**Comment by RyanK0902** [comment]
![CamScanner 07-06-2023 17 50_1](https://github.com/ThrunGroup/improving-xgboost/assets/79249503/83aa63aa-c6f2-401c-9bbb-4a702bdf200e)
This is the start of the weighted quantile sketch logic. Need to feed in a subset of the dataset into `batch` when calling `PushAdapterBatch`. Currently, `batch` is equivalent to entire dataset.

**Comment by siftheads1** [comment]
After the bin values are made with the procedure above, the values are stored in a instance of 'GHistIndexMatrix'(data structure that "converts gradient statistics to histogram index", according to the comment in the original source code).
The GHistIndexMatrix instance created above then becomes an attribute of DMatrix that holds a training data and later used for training.(IterativeDMatrix to be precise, but we believe this gets somehow converted to the DMatrix and gets passed to the gradient boosting module)
